core/domain/prod_validation_jobs_one_off.py

```diff
@@ -0,0 +1,156 @@
+# coding: utf-8
+#
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""One-off jobs for validating prod models."""
+from core import jobs
+from core.platform import models
+
+
+(
+    user_models, exp_models, collection_models,
+    feedback_models) = models.Registry.import_models([
+        models.NAMES.user, models.NAMES.exploration,
+        models.NAMES.collection, models.NAMES.feedback])
+datastore_services = models.Registry.import_datastore_services()
+
+
+def has_undeleted_model_for_id(model_class, model_id):
+    """Checks there's an existing nondeleted model for given model_id.
+
+    Args:
+        model_class: BaseModel. Class of the model.
+        model_id: str. Id of the model.
+
+    Returns:
+        bool. True if there exists a model with given model_id.
+    """
+    model = model_class.get_by_id(model_id)
+    return model is not None and model.deleted is False
+
+
+class BaseModelValidator(object):
+    """Base class for validating models."""
+
+    errors = {}
+
+    @classmethod
+    def _external_id_relationships(cls):
+        """Defines a mapping of external id to model class.
+
+        Returns a dictionary of which keys are field of the model to validate
+        and values are the corresponding model class. Checks the entries in that
+        field of the model should be valid non-deleted instances of the given
+        model classes.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def _validate_external_id_relationships(cls, item):
+        """Check whether the external id properties on the model correspond
+        to valid instances.
+
+        Args:
+            item: UserSubscriptionsModel. Entity to validate.
```

Comment:
Why is this specific model in BaseModelValidator (which should presumably be a general class)?